### ISSUE TITLE
Subclassing `Schema` loses any registered pagination connection classes

### ISSUE THREAD

**Issue by joelvh** [rmosolgo/graphql-ruby#2743]
I've found that using custom pagination connections doesn't work when `Schema` is subclassed, but does when using directly.

Here's an example of my `Schema`.

```ruby
class Schema < GraphQL::Schema
  use(GraphQL::Pagination::Connections)

  connections.add(CustomRelation, CustomRelationConnection)
end
```

When subclassing, such as `Class.new(Schema)`, I got this error: `RuntimeError: No connection implementation to wrap CustomRelation (#<CustomRelation:0x00007fb2f41efe70>)`

When using `Schema` directly, it works.

I should note some behavior I experienced before upgrading to the new interpreter and class-based types in order to try and fix this. Previously, I seemed to get the same error regardless of subclass or using the class directly. While debugging, I noticed that the `Schema#connections` was `nil` on the schema instance, but it *did* have a configuration object on the schema class (`Schema.connections`).

In summary, it works without subclassing using the new interpreter and class-based types.

**Comment by rmosolgo** [comment]
Thanks for sharing the details on this. I bet the problem is that `.connections` is not at all inheritance-aware, a shortcoming in my initial implementation here: 

https://github.com/rmosolgo/graphql-ruby/blob/e33a84dce6e71d4c3b0929790e826dc0c66f4004/lib/graphql/schema.rb#L945-L950

It _should_ fall back to `find_inherited_value`, similar to max_page_size:
 
https://github.com/rmosolgo/graphql-ruby/blob/e33a84dce6e71d4c3b0929790e826dc0c66f4004/lib/graphql/schema.rb#L1136-L1142

Feel free to take a try at the fix if you like, otherwise I'll take a look soon.

**Comment by joelvh** [comment]
Hi @rmosolgo thanks for the pointers. I’ll see if I can come up with a solution

**Comment by joelvh** [comment]
oh Wow! Just realized you fixed this - thanks!
